llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -226,10 +227,29 @@ static void saveResults(std::vector<std::unique_ptr<Module>> &ResModules,
     ++NumOfFile;
   }
 
-  if (OutputIRFilesList != "-")
+  if (OutputIRFilesList != "-") {
+    // TODO: Figure out what can be added to output list if there are no kernels
+    // in the input module
+    // Just pass input module to next tools if there was nothing to split
+    if (IRFilesList.empty())
+      IRFilesList =
+          (Twine(InputFilename) + Twine("\n")).str();
     writeToFile(OutputIRFilesList, IRFilesList);
-  if (OutputTxtFilesList != "-")
+  }
+  if (OutputTxtFilesList != "-") {
+    // TODO: Figure out what can be added to output list if there are no kernels
+    // in the input module
```

Comment:
What exactly will the RT gain from this info?

Reply:
The RT will gain nothing. But all actions in the driver should have outputs, so `sycl-post-link` should give some output even there is no kernels.